### PR TITLE
Make the storage-users propagator configurable

### PR DESCRIPTION
**Pull request by aduffeck**
Depends on https://github.com/cs3org/reva/pull/4070

### CODE REVIEW

File: services/storage-users/pkg/config/config.go [hunk 1]
```diff
@@ -99,9 +99,16 @@ type Drivers struct {
 	Local LocalDriver `yaml:",omitempty"` // not supported by the oCIS product, therefore not part of docs
 }
 
+// AsyncPropagatorOptions configures the async propagator
+type AsyncPropagatorOptions struct {
+	PropagationDelay time.Duration `yaml:"propagation_delay" env:"STORAGE_USERS_ASYNC_PROPAGATOR_PROPAGATION_DELAY" desc:"The delay between a change to tree and the start of the according treesize and treetime propagation."`
+}
```
Comment (kobergj) [inline services/storage-users/pkg/config/config.go:105]: @mmattel do we need to make this clearer?

**Review by kobergj — COMMENTED** [review]
LGTM :+1: can  be merged if @mmattel approves.

File: services/storage-users/pkg/config/config.go (hunk 1 above)
Comment (mmattel) [reply services/storage-users/pkg/config/config.go:105]: I will check today, but I need to finish some private stuff.

File: services/storage-users/pkg/config/config.go [hunk 2]
```diff
@@ -99,9 +99,16 @@ type Drivers struct {
 	Local LocalDriver `yaml:",omitempty"` // not supported by the oCIS product, therefore not part of docs
 }
 
+// AsyncPropagatorOptions configures the async propagator
+type AsyncPropagatorOptions struct {
+	PropagationDelay time.Duration `yaml:"propagation_delay" env:"STORAGE_USERS_ASYNC_PROPAGATOR_PROPAGATION_DELAY" desc:"The delay between a change to tree and the start of the according treesize and treetime propagation."`
```
Comment (mmattel) [inline services/storage-users/pkg/config/config.go:104]:
```suggestion
	PropagationDelay time.Duration `yaml:"propagation_delay" env:"STORAGE_USERS_ASYNC_PROPAGATOR_PROPAGATION_DELAY" desc:"The delay in seconds between a change made to a to tree and the propagation start on treesize and treetime. Multiple propagations are computed to a single one."`
```

File: services/storage-users/pkg/config/config.go (hunk 1 above)
Comment (mmattel) [reply services/storage-users/pkg/config/config.go:105]: Done

**Pull request merged**